cmd/cored/main.go

```diff
@@ -419,11 +419,8 @@ type remoteSigner struct {
 	Key    ed25519.PublicKey
 }
 
-func (s *remoteSigner) SignBlock(ctx context.Context, b *legacy.Block) (signature []byte, err error) {
-	// TODO(kr): We might end up serializing b multiple
-	// times in multiple calls to different remoteSigners.
-	// Maybe optimize that if it makes a difference.
-	err = s.Client.Call(ctx, "/rpc/signer/sign-block", b, &signature)
+func (s *remoteSigner) SignBlock(ctx context.Context, marshalledBlock []byte) (signature []byte, err error) {
+	err = s.Client.Call(ctx, "/rpc/signer/sign-block", string(marshalledBlock), &signature)
```

Comment:
Ah, I think this function signature is confusing. Maybe `marshalledBlock` should be a string?

I think this should only change the Go interfaces. Previously, Call would json.Marshal the block, calling Block.MarshalText. The resulting request body would be a json string contianing the hex-encoded bytes:
```
"abcdef12..."
```

Now, SignBlock takes the output of Block.MarshalText directly. I think it can't be a json.RawMessage because it doesn't have the json string quotes yet.